scripts/visualize/visualize-hydrograph.R

```diff
@@ -20,29 +20,37 @@ visualize.plot_hydrographTotal <- function(viz){
   
   total_svg <- svglite::xmlSVG({
     par(omi=c(0,0,0,0), mai=c(0.3,0,0,0),las=1,xaxs="i")
-    plot(data_hydrograph, type='l', axes=F, ann=F, xaxt="n")
+    plot(data_hydrograph, type='l', axes=F, ann=F, xaxt="n",
+         xlim=c(0, pixelWidth), ylim=c(0, pixelHeight))
     axis(side=1, at=at, labels=labels, cex.axis=0.6)
   }, height=pixelHeight/72, width=pixelWidth/72)
   total_svg <- clean_up_svg(total_svg, viz)
   
-  svgaxis <- xml_children(total_svg)
-  
-  hydrograph <- xml_children(total_svg)[3]
   xml_set_attr(total_svg, 'fill', "none")
   xml_set_attr(total_svg, 'stroke', "black")
 
-  for(yr in names(rects)){
-    rect <- xml_add_child(total_svg, 'rect', 
-                          x=rects[[yr]][['x']], 
-                          width=rects[[yr]][['width']], 
-                          height=rects[[yr]][['height']], 
-                          id=yr)
-    xml_set_attr(rect, 'fill', "blue")
-    xml_set_attr(rect, 'stroke', "blue")
-    xml_set_attr(rect, 'stroke-width', "0.5")
-    xml_set_attr(rect, 'opacity', "0")
-    xml_set_attr(rect, 'onmouseover', "evt.target.setAttribute('opacity', '0.5');")
-    xml_set_attr(rect, 'onmouseout', "evt.target.setAttribute('opacity', '0');")
+  # for(yr in names(rectangles)){
+  for(yr in names(rectangles)[1]){
+    rect_svg_all <- svglite::xmlSVG({
+      par(omi=c(0,0,0,0), mai=c(0.3,0,0,0),las=1,xaxs="i")
+      plot.new()
+      rect(xleft=rectangles[[yr]][['xleft']],
+         xright=rectangles[[yr]][['xright']],
+         ybottom=0, ytop=pixelHeight, 
+         border="blue", col="blue",
+         ann=F, xaxt="n",
+         xlim=c(0, pixelWidth), ylim=c(0, pixelHeight))
+    },  height=pixelHeight/72, width=pixelWidth/72)
+  
+    rect_svg <- xml_children(rect_svg_all)[4]
```

Comment:
Can someone paraphrase why this went from `rect`s to capturing the plot and decorating after?